website/docs/feature-flag-tutorials/use-cases/ai.md

```diff
@@ -0,0 +1,223 @@
+
+To implement this, we'll need to:
+
+1. Set up a feature flag provider (we'll use Unleash).
+2. Replace our static variant selection with feature flag calls.
+3. Use the feature flag in our code to determine which AI model and settings to use for each request.
+
+### Install a local feature flag provider
+
+In this section, we’ll install Unleash, run the instance locally, log in, and create a feature flag. If you prefer, you can use other tools instead of Unleash, but you’ll need to update the code accordingly.
+
+Use Git to clone the Unleash repository and Docker to build and run it. Open a terminal window and run the following commands:
+
+```sh
+git clone https://github.com/unleash/unleash.git
+cd unleash
+docker compose up -d
+```
+
+You now have Unleash installed on your machine and running in the background. You can access this instance in your web browser at [http://localhost:4242](http://localhost:4242).
+
+Log in to the platform using these credentials:
+
+```
+Username: admin
+Password: unleash4all
+```
+
+Click **New feature flag** to create a new feature flag.
+
+![Create a new feature flag](../ruby/new-ff.png)
+
+Create a new feature flag called `gpt-version`.
+
+After that, and this the most important part, we need to add a variant to our feature flag.
+
+What we'll do is hold all the model configurations in the feature flag variants.
+
+So, we can take these model configurations:
+
+```javascript
+const variants = {
+    variant1: {
+        model: "gpt-3.5-turbo",
+        max_tokens: 4096,
+        temperature: 1,
+    },
+    variant2: {
+        model: "gpt-4-turbo",
+        max_tokens: 4096,
+        temperature: 1,
+    },
+    variant3: {
+        model: "gpt-4-vision-preview",
+        max_tokens: 4096,
+        temperature: 1,
+    },
+};
+```
+
+And add them into Unleash.
+
+![a variant with parameters for an OpenAI model](./model-variant.png)
+
+## Querying AI feature flags
+
+Now let's go back to the code and grab our AI config from the feature flag that we just created.
+
+First, install the Unleash Node.js client:
+
+```sh
+npm install unleash-client
+```
+
+Now, let's modify our `+server.js` file to use Unleash:
+
+```javascript
+import { initialize } from "unleash-client";
+import { createOpenAI } from "@ai-sdk/openai";
+import { streamText } from "ai";
+import { env } from "$env/dynamic/private";
+
+const openai = createOpenAI({
+    apiKey: env.OPENAI_API_KEY ?? "",
+});
+
+const unleash = initialize({
+    url: "http://localhost:4242/api/",
+    appName: "my-ai-app",
+    customHeaders: { Authorization: env.UNLEASH_API_KEY ?? "" },
+});
+
+export const POST = async ({ request }) => {
+    const { messages } = await request.json();
+
+    // Get the feature flag variant
+    const variant = unleash.getVariant("gpt-versions");
```

Comment:
It should be `gpt-version`, not `gpt-versions`, right? I suggest going through the guide yourself to double-check everything works correctly and to ensure we didn’t miss small details like this.

```suggestion
    const variant = unleash.getVariant("gpt-version");
```